addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/thermostat/NATherm1Handler.java

```diff
@@ -94,32 +115,102 @@ protected State getNAThingProperty(String channelId) {
                     if (setpoint != null) {
                         Integer endTime = setpoint.getSetpointEndtime();
                         if (endTime == null) {
-                            endTime = getNextSchedule(module.getThermProgramList());
+                            endTime = getNextProgramTime(module.getThermProgramList());
                         }
                         return toDateTimeType(endTime);
                     }
                     return UnDefType.NULL;
                 }
                 return UnDefType.UNDEF;
             }
-            case CHANNEL_SETPOINT_MODE: {
-                return module != null ? module.getSetpoint() != null
-                        ? ChannelTypeUtils.toStringType(module.getSetpoint().getSetpointMode())
-                        : UnDefType.NULL : UnDefType.UNDEF;
+            case CHANNEL_SETPOINT_MODE:
+                return getSetpoint();
+            case CHANNEL_PLANNING: {
+                String currentPlanning = "-";
+                if (module != null) {
+                    for (NAThermProgram program : module.getThermProgramList()) {
+                        if (program.getSelected() == Boolean.TRUE) {
+                            currentPlanning = program.getProgramId();
+                        }
+                    }
+                    return toStringType(currentPlanning);
+                }
             }
         }
         return super.getNAThingProperty(channelId);
     }
 
-    private int getNextSchedule(List<NAThermProgram> thermProgramList) {
+    private State getSetpoint() {
+        return module != null
+                ? module.getSetpoint() != null ? toStringType(module.getSetpoint().getSetpointMode()) : UnDefType.NULL
+                : UnDefType.UNDEF;
+    }
+
+    private State getCurrentSetpoint() {
+        if (module != null && module.getSetpoint() != null) {
+            NASetpoint setPoint = module.getSetpoint();
+            String currentMode = setPoint.getSetpointMode();
+
+            NAThermProgram currentProgram = module.getThermProgramList().stream()
+                    .filter(p -> p.getSelected() != null && p.getSelected()).findFirst().get();
```

Comment:
Is there always at least one option selected? Because doing an unchecked `get` call is in my idea a warning.

Reply:
And if your get returns, it is always non-null

Reply:
As far as I understand the Netatmo API, you can not be without a selected planning, there is always a default one.

Reply:
so, you're right, the null test is pointless...if I'm right :)